pkg/gather/clusterconfig/operators.go

```diff
@@ -104,16 +109,116 @@ func gatherClusterOperators(ctx context.Context, configClient configv1client.Con
+	klog.V(2).Infof("Maximum buffer size: %v bytes", bufferSize)
+	buf := bytes.NewBuffer(make([]byte, 0, bufferSize))
+
+	// Fetch previous and current container logs
+	for _, isPrevious := range []bool{true, false} {
+		for _, pod := range pods {
+			clog := collectPodContainerLogs(ctx, client, pod, isPrevious, buf, bufferSize)
+			if len(clog) > 0 {
+				records = append(records, clog...)
+			}
+		}
+	}
+
+	return records
+}
+
+// isPodRestarted checks if pod was restarted by testing its container's restart count status is bigger than zero
+func isPodRestarted(pod *corev1.Pod) bool {
+	// pods that have containers that have terminated with non-zero exit codes are considered failure
+	for _, status := range pod.Status.InitContainerStatuses {
+		if status.RestartCount > 0 {
+			return true
+		}
+	}
+	for _, status := range pod.Status.ContainerStatuses {
+		if status.RestartCount > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+// collectPodContainerLogs generates the previous and current log reports for pod containers using the k8s API response
+func collectPodContainerLogs(ctx context.Context, client corev1client.CoreV1Interface, pod *corev1.Pod, isPrevious bool, buf *bytes.Buffer, bufferSize int64) []record.Record {
+	var records []record.Record
+
+	allContainers := append(pod.Spec.InitContainers, pod.Spec.Containers...)
+	for _, c := range allContainers {
+		// only grab previous log if the pod is restarted
+		if isPrevious && !isPodRestarted(pod) {
+			continue
+		}
+
+		logName := fmt.Sprintf("%s_current.log", c.Name)
+		if isPrevious {
+			logName = fmt.Sprintf("%s_previous.log", c.Name)
+		}
+
+		buf.Reset()
+		klog.V(2).Infof("Fetching logs for %s container %s pod in namespace %s (previous: %v).", c.Name, pod.Name, pod.Namespace, isPrevious)
+		// Fetch container logs and continue on error
+		err := fetchPodContainerLog(ctx, client, pod, buf, c.Name, isPrevious, &bufferSize)
+		if err != nil {
+			klog.V(2).Infof("Error: %q", err)
+			continue
+		}
+
+		// Do not record empty records
+		if buf.Len() == 0 {
+			continue
+		}
+		records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/logs/%s/%s", pod.Namespace, pod.Name, logName), Item: marshal.Raw{Str: buf.String()}})
+	}
+
+	return records
+}
+
+// clusterOperatorsRecords generates the cluster operator records
+func clusterOperatorsRecords(ctx context.Context, items []configv1.ClusterOperator, dynamicClient dynamic.Interface, discoveryClient discovery.DiscoveryInterface) []record.Record {
 	resVer, _ := getOperatorResourcesVersions(discoveryClient)
-	records := make([]record.Record, 0, len(config.Items))
-	for idx, co := range config.Items {
+	records := make([]record.Record, 0, len(items))
+
+	for idx, co := range items {
 		records = append(records, record.Record{
-			Name: fmt.Sprintf("config/clusteroperator/%s", config.Items[idx].Name),
-			Item: record.JSONMarshaller{Object: &config.Items[idx]},
+			Name: fmt.Sprintf("config/clusteroperator/%s", items[idx].Name),
```

Comment:
You could just use `co` instead of `items[idx]` here, right? But I get it for the sake of consistency with the line below.

Reply:
True.